apps/web/composables/useInitialSetup/useInitialSetup.ts

```diff
@@ -52,7 +52,16 @@ const setInitialDataSSR: SetInitialData = async () => {
       setUser(data.value.data.session as SessionResult);
       setCart(data.value.data.session?.basket as Cart);
       setCategoryTree(data.value.data.categories);
-      setWishlistItemIds(data.value.data.session?.basket?.itemWishListIds || []);
+      const wishlistIdsFromSession = data.value.data.session?.basket?.itemWishListIds || [];
+      const wishlistIdsJSON = JSON.parse(JSON.stringify(wishlistIdsFromSession));
+      const wishlistIds = [];
+      const wishlistVariationIds = [];
+      for (const wishlistObject of wishlistIdsJSON) {
+        wishlistIds.push(wishlistObject.variationId);
+        wishlistVariationIds.push(wishlistObject);
+      }
+      setWishlistItemIds(wishlistIds || []);
+      setWishlistVariationIds(wishlistVariationIds || []);
```

Comment:
Why couldn't `setWishlistItemIds` method be modified to account for the new parameter ? We are basically doing the same thing now in 2 separate methods.

Do we need to stringify, parse and then iterate over the response again ? Why doesn't it work without doing that ?